internal/cli/uploadcmd/archive.go

```diff
@@ -107,6 +120,10 @@ func archiveModule(root string) (io.Reader, error) {
 			return err
 		}
 
+		// manually close here after each file operation; defering would cause each file close
+		// to wait until all operations have completed.
+		data.Close()
```

Comment:
Ah thanks for catching this! :pray:
Why wouldn't defer work here though?

Reply:
I've got `too many open files` exceptions when compressing lots of file here. Also the defer relates to the resulting tar archive - this is for the data to compress.